speculation-rules.bs

```diff
@@ -275,10 +389,23 @@ A <dfn>prerender candidate</dfn> is a [=struct=] with the following [=struct/ite
       1. &#x231B; If |rule|'s [=speculation rule/requirements=] [=set/contains=] "`anonymous-client-ip-when-cross-origin`", set |anonymizationPolicy| to a [=cross-origin prefetch IP anonymization policy=] whose [=cross-origin prefetch IP anonymization policy/origin=] is |document|'s [=Document/origin=].
       1. &#x231B; [=list/For each=] |url| of |rule|'s [=speculation rule/URLs=]:
         1. &#x231B; [=list/Append=] a [=prefetch candidate=] with [=prefetch candidate/URL=] |url| and [=prefetch candidate/anonymization policy=] |anonymizationPolicy| to |prefetchCandidates|.
+      1. &#x231B; If |rule|'s [=speculation rule/predicate=] is not null, then:
+        1. &#x231B; Let |links| be the result of [=finding matching links=] given |document| and |rule|'s [=speculation rule/predicate=].
+        1. &#x231B; [=list/For each=] |link| of |links|:
+          1. &#x231B; Let |href| be the result of running |link|'s {{HTMLHyperlinkElementUtils/href}} getter steps.
+          1. &#x231B; [=list/Append=] a [=prefetch candidate=] with [=prefetch candidate/URL=] |href|, and [=prefetch candidate/anonymization policy=] |anonymizationPolicy| to |prefetchCandidates|.
     1. &#x231B; [=list/For each=] |rule| of |ruleSet|'s [=speculation rule set/prerender rules=]:
       1. &#x231B; [=list/For each=] |url| of |rule|'s [=speculation rule/URLs=]:
         1. &#x231B; Let |prerenderCandidate| be a new [=prerender candidate=] whose [=prerender candidate/URL=] is |url| and [=prerender candidate/target browsing context name hint=] is |rule|'s [=speculation rule/target browsing context name hint=].
         1. &#x231B; [=list/Append=] |prerenderCandidate| to |prerenderCandidates|.
+      1. &#x231B; If |rule|'s [=speculation rule/predicate=] is not null, then:
+        1. &#x231B; Let |links| be the result of [=finding matching links=] given |document| and |rule|'s [=speculation rule/predicate=].
+        1. &#x231B; [=list/For each=] |link| of |links|:
+          1. &#x231B; Let |href| be the result of running |link|'s {{HTMLHyperlinkElementUtils/href}} getter steps.
+          1. &#x231B; Let |target| be |rule|'s [=speculation rule/target browsing context name hint=].
+          1. &#x231B; If |target| is null, set it to the result of [=getting an element's target=] given |link|.
```

Comment:
Hmm, this is making me wonder if we should use `<base target>` for list rules too, just like we already use `<base href>`!

Reply:
That's a valid question. I don't think the answer is obvious -- for links, we should respect whatever target the link would have, but for a list rule we don't know whether a `<base target>` respecting link navigation will be used or not.

While `<base href>` defines how all URLs should be interpreted throughout the document (and so clearly must apply), `<base target>` merely provides a default value for `<a>`, `<form>`, and a handful of other cases.

As another example, navigation by `location = ...` would respect `<base href>` but not `<base target>`, while link navigation and form submission would affect both.

Reply:
The `location` example convinced me; thanks!